src/models/index.js

```diff
@@ -0,0 +1,146 @@
+
+let db;
+
+function initialize() {
+    let dbReq = indexedDB.open('publiclabDB');
+
+    // Fires when upgrade needed
+    dbReq.onupgradeneeded = function(event) {
+        // Set the db variable to our database so we can use it
+        db = event.target.result;
+
+        // Create an object store named toolbox, or retrieve it if it already exists.
+        // Object stores in databases are where data are stored.
+        let toolbox;
+        if (!db.objectStoreNames.contains('toolbox')) {
+            toolbox = db.createObjectStore('toolbox', {autoIncrement: true});
+        } else {
+            toolbox = db.transaction("toolbox", "readwrite").objectStore("toolbox");
+        }
+
+        // If there isn't already a KEY index, make one so we can query toolbox
+        // by their KEY
+        if (!toolbox.indexNames.contains('key')) {
+            toolbox.createIndex('key', 'key', { unique: true });
+        }else {
+            console.log("KEY index is already created");
+        }
+
+        // If there isn't already a CONTENT index, make one so we can query toolbox
+        // by their CONTENT
+        if (!toolbox.indexNames.contains('content')) {
+            toolbox.createIndex('content', 'content', { unique: false });
+        } else {
+            console.log("content index is already created");
+        }
+    }
+
+    // Fires once the database is opened (and onupgradeneeded completes, if
+    // onupgradeneeded was called)
+    dbReq.onsuccess = function(event) {
+        // Set the db variable to our database so we can use it
+        db = event.target.result;
+    }
+    
+    // Fires when we can't open the database
+    dbReq.onerror = function(event) {
+        console.log('error opening database');
+    }
+}
+
+// We make sure that initialize function gets fired once the DOM content is loaded
+window.addEventListener('DOMContentLoaded', initialize());
+
+
+function saveContentToDb(queryKey, queryContent) {
```

Comment:
This is looking amazing. One way to begin making it more readable would be to break these into separate files and use `require()` where possible.

Reply:
Yeah, that will make it more clear and more specific.